Extensions/P2P/B_p2ptools.js

```diff
@@ -0,0 +1,308 @@
+// @ts-check
+/// <reference path="peerjs" />
+
+/**
+ * Tools for p2p multiplayer.
+ * @namespace
+ */
+gdjs.evtTools.p2p = {
+  /**
+   * The peer to peer configuration.
+   */
+  peerConfig: {debug:1}, // Enable logging of critical errors
+
+  /**
+   * The p2p client.
+   * @type {?Peer}
+   */
+  peer: null,
+
+  /**
+   * All connected p2p clients, keyed by their id.
+   */
+  connections: {},
+
+  /**
+   * Contains a list of events triggered by other p2p clients.
+   */
+  triggeredEvents: {},
+
+  /**
+   * Contains the latest data sent with each event.
+   */
+  lastEventData: {},
+
+  /**
+   * Tells how to handle an event (with or without data loss)
+   */
+  eventHandling: {},
+
+  /**
+   * True if PeerJS is initialized and ready.
+   */
+  ready: false,
+
+  /**
+   * True if an error occured.
+   */
+  error: false,
+
+  /**
+   * Last error's message.
+   */
+  lastError: "",
+
+  /**
+   * True if a peer diconnected.
+   */
+  peerDisconnected: false,
```

Comment:
to show that this is valid only once, I usually like naming this with "just": `peerJustDisconnected`